src/config/config.h

```diff
@@ -194,6 +194,7 @@ struct Config {
   mutable std::mutex backup_mu_;
 
  public:
+  const std::string NodesFilePath();
```

Comment:
Did you mean `std::string NodesFilePath() const;`? Because, in my opinion, it makes no sense to return std::string by value and apply const-modifier to that value.